src/tests/Common/XUnitWrapperGenerator/ITestInfo.cs

```diff
@@ -0,0 +1,139 @@
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+
+namespace XUnitWrapperGenerator;
+
+interface ITestInfo
+{
+    string ExecutionStatement { get; }
+}
+
+sealed class BasicTestMethod : ITestInfo
+{
+    public BasicTestMethod(IMethodSymbol method, string externAlias, ImmutableArray<string> arguments = default)
+    {
+        var args = arguments.IsDefaultOrEmpty ? "" : string.Join(", ", arguments);
+        string containingType = method.ContainingType.ToDisplayString(XUnitWrapperGenerator.FullyQualifiedWithoutGlobalNamespace);
+        if (method.IsStatic)
+        {
+            ExecutionStatement = $"{externAlias}::{containingType}.{method.Name}({args});";
+        }
+        else
+        {
+            ExecutionStatement = $"using ({externAlias}::{containingType} obj = new()) obj.{method.Name}({args});";
+        }
+    }
+
+    public string ExecutionStatement { get; }
+
+    public override bool Equals(object obj)
+    {
+        return obj is BasicTestMethod other && ExecutionStatement == other.ExecutionStatement;
+    }
+}
+
+sealed class ConditionalTest : ITestInfo
+{
+    public ConditionalTest(ITestInfo innerTest, string condition)
+    {
+        ExecutionStatement = $"if ({condition}) {{ {innerTest.ExecutionStatement} }}";
+    }
+
+    public string ExecutionStatement { get; }
+
+    public override bool Equals(object obj)
+    {
+        return obj is ConditionalTest other && ExecutionStatement == other.ExecutionStatement;
+    }
+}
+
+sealed class PlatformSpecificTest : ITestInfo
+{
+    public PlatformSpecificTest(ITestInfo innerTest, Xunit.TestPlatforms platform)
+    {
+        List<string> platformCheckConditions = new();
+        if (platform.HasFlag(Xunit.TestPlatforms.Windows))
+        {
+            platformCheckConditions.Add("global::System.OperatingSystem.IsWindows()");
+        }
+        if (platform.HasFlag(Xunit.TestPlatforms.Linux))
+        {
+            platformCheckConditions.Add("global::System.OperatingSystem.IsLinux()");
+        }
+        if (platform.HasFlag(Xunit.TestPlatforms.OSX))
+        {
+            platformCheckConditions.Add("global::System.OperatingSystem.IsMacOS()");
+        }
+        if (platform.HasFlag(Xunit.TestPlatforms.illumos))
+        {
+            platformCheckConditions.Add(@"global::System.OperatingSystem.IsOSPlatform(""illumos"")");
+        }
+        if (platform.HasFlag(Xunit.TestPlatforms.Solaris))
+        {
+            platformCheckConditions.Add(@"global::System.OperatingSystem.IsOSPlatform(""Solaris"")");
+        }
+        if (platform.HasFlag(Xunit.TestPlatforms.Android))
+        {
+            platformCheckConditions.Add("global::System.OperatingSystem.IsAndroid()");
+        }
+        if (platform.HasFlag(Xunit.TestPlatforms.iOS))
+        {
+            platformCheckConditions.Add("global::System.OperatingSystem.IsIOS()");
```

Comment:
iOS is a special case since `OperatingSystem.IsIOS()` returns true on both iOS and MacCatalyst, you'll need to check for `!OperatingSystem.IsIOS()` like we do in arcade: https://github.com/dotnet/arcade/blob/c93266eb7c9b40a63f9eae2ce00c58b7cc49f721/src/Microsoft.DotNet.XUnitExtensions/src/DiscovererHelpers.cs#L28

Reply:
```suggestion
            platformCheckConditions.Add("(global::System.OperatingSystem.IsIOS() && !global::System.OperatingSystem.IsMacCatalyst())");
```